File: src/wsTripXML/Code/OTAClasses/IssueMCOModels.vb
Imports System.Runtime.Serialization

Namespace wsTravelTalk.wmIssueMCOModels
    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    <Serializable>
    Public Class MCO
#Region "Request Fields"
        Public Property ErrorMessage As String
        Public Property Amount As String
        Public Property PassengerNumber As String
        Public Property PaxNumber As String
        Public Property PassengerName As String
        Public Property PaxType As String
        Public Property TicketingAirlineCode As String
        Public Property [To] As String
        Public Property AT As String
        Public Property Tax As String
        Public Property TaxCode As String
        Public Property TypeOfService() As String
        Public Property PQNumber As Integer
        Public Property TourNumber As String
        Public Property Endorsements() As String
        Public Property Remarks() As String
        Public Property TicketNumber As String
        Public Property CurrencyCode As String
        Public Property EquivelentAmount As String
        Public Property EquivCurrencyCode As String
        Public Property RateOfExchange As String
        Public Property CommissionPercentage As String
        Public Property CommissionAmount As String
        Public Property InternationalItin As Boolean

        'Form of payments
        Public Property CASH As Boolean
        Public Property Check As Boolean
        Public Property Cheque As Boolean
        Public Property CreditCard As String
        Public Property Expiration As String
        Public Property ApprovalCode As String
        Public Property Vendor As String
#End Region
    End Class
    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    <KnownType(GetType(MCO))>
    <Serializable>
    Public Class MCOMask
        Inherits MCO
        Public Sub New()
            ErrorMessage = String.Empty
            AT = "NYC"
            StatementInformation = String.Empty
            TourNumber = String.Empty
            TicketNumber = String.Empty
            Ignore = False
            CurrencyCode = "USD"
            EquivelentAmount = String.Empty
            EquivCurrencyCode = String.Empty
            RateOfExchange = String.Empty
            Total = String.Empty
            CommissionPercentage = String.Empty
            CommissionAmount = String.Empty
            InternationalItin = False
            CASH = False
            Cheque = False
            SelfSale = False
            CreditCard = " "
            Expiration = " "
            ApprovalCode = " "
            isAX = False
            isSuppressed = False
            SGR = " "
            Other = " "
            GTR = " "
            IgnoreTJR = False
            IgnoreCouponPrint = False
            DisplayPrevMask = False
            IssueDocuments = True
            IsDone = False
            Id = String.Empty
        End Sub

#Region "MCO Creat Fields"
        Public Property Total As String
        Public Property Id As String
        Public Property StatementInformation As String
        Public Property Ignore As Boolean
        Public Property SelfSale As Boolean
        Public Property isAX As Boolean
        Public Property isSuppressed As Boolean
        Public Property SGR As String
        Public Property Other As String
        Public Property GTR As String
        Public Property IgnoreTJR As Boolean
        Public Property IgnoreCouponPrint As Boolean
        Public Property DisplayPrevMask As Boolean
        Public Property IssueDocuments As Boolean
        Public Property IsDone As Boolean
#End Region

#Region "MCO Exchange Fields"
        Public Property MCONumber As String
        Public Property ChangeFeeAmount As String
        Public Property IsEMDFee As Boolean
        Public Property TaxComparison As Boolean
        Public Property NewTktCommAmount As String
        Public Property AddCollCommAmount As String
        Public Property Waiver As String
        Public Property IsModifyBAG As Boolean
        Public Property IsEndorsmentsUpdate As Boolean
        Public Property ManualApproval As String
        Public Property IsTicketing As Boolean
        Public Property IsRetain As Boolean
        Public Property IsPrevious As Boolean
        Public Property IsNext As Boolean
        Public Property IsQuit As Boolean
#End Region

    End Class

    '<remarks/>
    Public Enum TransactionStatusCode

        '<remarks/>
        Start

        '<remarks/>
        [End]

        '<remarks/>
        Rollback

        '<remarks/>
        InSeries
    End Enum

    '<remarks/>
    Public Enum Target
        '<remarks/>
        Test
        '<remarks/>
        Production
    End Enum

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    Public Class POS

        '<remarks/>
        Public Source As Source

        '<remarks/>
        Public TPA_Extensions As TPA_Extensions
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    Public Class Source

        '<remarks/>
        Public RequestorID As RequestorID

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public AgentSine As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public PseudoCityCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public ISOCountry As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public ISOCurrency As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public AgentDutyCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public AirlineVendorID As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public AirportCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public FirstDepartPoint As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public ERSP_UserID As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    Public Class RequestorID

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute(DataType:="anyURI")>
        Public URL As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public Type As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public Instance As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public ID As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public ID_Context As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    Public Class UniqueID

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public ID As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    Public Class Provider

        '<remarks/>
        Public Name As String

        '<remarks/>
        Public System As String

        '<remarks/>
        Public Userid As String

        '<remarks/>
        Public Password As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    Public Class Success
    End Class
End Namespace

File: src/wsTripXML/Code/OTAClasses/IssueMCOModels.cs
using System;
using System.Runtime.Serialization;

namespace wsTripXML.wsTravelTalk.wmIssueMCOModels
{
    // <remarks/>
    [System.Xml.Serialization.XmlRoot(IsNullable = false)]
    [Serializable]
    public class MCO
    {
        #region Request Fields
        public string ErrorMessage { get; set; }
        public string Amount { get; set; }
        public string PassengerNumber { get; set; }
        public string PaxNumber { get; set; }
        public string PassengerName { get; set; }
        public string PaxType { get; set; }
        public string TicketingAirlineCode { get; set; }
        public string To { get; set; }
        public string AT { get; set; }
        public string Tax { get; set; }
        public string TaxCode { get; set; }
        public string TypeOfService { get; set; }
        public int PQNumber { get; set; }
        public string TourNumber { get; set; }
        public string Endorsements { get; set; }
        public string Remarks { get; set; }
        public string TicketNumber { get; set; }
        public string CurrencyCode { get; set; }
        public string EquivelentAmount { get; set; }
        public string EquivCurrencyCode { get; set; }
        public string RateOfExchange { get; set; }
        public string CommissionPercentage { get; set; }
        public string CommissionAmount { get; set; }
        public bool InternationalItin { get; set; }

        // Form of payments
        public bool CASH { get; set; }
        public bool Check { get; set; }
        public bool Cheque { get; set; }
        public string CreditCard { get; set; }
        public string Expiration { get; set; }
        public string ApprovalCode { get; set; }
        public string Vendor { get; set; }
        #endregion
    }
    // <remarks/>
    [System.Xml.Serialization.XmlRoot(IsNullable = false)]
    [KnownType(typeof(MCO))]
    [Serializable]
    public class MCOMask : MCO
    {
        public MCOMask()
        {
            ErrorMessage = string.Empty;
            AT = "NYC";
            StatementInformation = string.Empty;
            TourNumber = string.Empty;
            TicketNumber = string.Empty;
            Ignore = false;
            CurrencyCode = "USD";
            EquivelentAmount = string.Empty;
            EquivCurrencyCode = string.Empty;
            RateOfExchange = string.Empty;
            Total = string.Empty;
            CommissionPercentage = string.Empty;
            CommissionAmount = string.Empty;
            InternationalItin = false;
            CASH = false;
            Cheque = false;
            SelfSale = false;
            CreditCard = " ";
            Expiration = " ";
            ApprovalCode = " ";
            isAX = false;
            isSuppressed = false;
            SGR = " ";
            Other = " ";
            GTR = " ";
            IgnoreTJR = false;
            IgnoreCouponPrint = false;
            DisplayPrevMask = false;
            IssueDocuments = true;
            IsDone = false;
            Id = string.Empty;
        }

        #region MCO Creat Fields
        public string Total { get; set; }
        public string Id { get; set; }
        public string StatementInformation { get; set; }
        public bool Ignore { get; set; }
        public bool SelfSale { get; set; }
        public bool isAX { get; set; }
        public bool isSuppressed { get; set; }
        public string SGR { get; set; }
        public string Other { get; set; }
        public string GTR { get; set; }
        public bool IgnoreTJR { get; set; }
        public bool IgnoreCouponPrint { get; set; }
        public bool DisplayPrevMask { get; set; }
        public bool IssueDocuments { get; set; }
        public bool IsDone { get; set; }
        #endregion

        #region MCO Exchange Fields
        public string MCONumber { get; set; }
        public string ChangeFeeAmount { get; set; }
        public bool IsEMDFee { get; set; }
        public bool TaxComparison { get; set; }
        public string NewTktCommAmount { get; set; }
        public string AddCollCommAmount { get; set; }
        public string Waiver { get; set; }
        public bool IsModifyBAG { get; set; }
        public bool IsEndorsmentsUpdate { get; set; }
        public string ManualApproval { get; set; }
        public bool IsTicketing { get; set; }
        public bool IsRetain { get; set; }
        public bool IsPrevious { get; set; }
        public bool IsNext { get; set; }
        public bool IsQuit { get; set; }
        #endregion

    }

    // <remarks/>
    public enum TransactionStatusCode
    {

        // <remarks/>
        Start,

        // <remarks/>
        End,

        // <remarks/>
        Rollback,

        // <remarks/>
        InSeries
    }

    // <remarks/>
    public enum Target
    {
        // <remarks/>
        Test,
        // <remarks/>
        Production
    }

    // <remarks/>
    [System.Xml.Serialization.XmlRoot(IsNullable = false)]
    public class POS
    {

        // <remarks/>
        public Source Source;

        // <remarks/>
        public TPA_Extensions TPA_Extensions;
    }

    // <remarks/>
    [System.Xml.Serialization.XmlRoot(IsNullable = false)]
    public class Source
    {

        // <remarks/>
        public RequestorID RequestorID;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string AgentSine;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string PseudoCityCode;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string ISOCountry;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string ISOCurrency;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string AgentDutyCode;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string AirlineVendorID;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string AirportCode;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string FirstDepartPoint;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string ERSP_UserID;
    }

    // <remarks/>
    [System.Xml.Serialization.XmlRoot(IsNullable = false)]
    public class RequestorID
    {

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute(DataType = "anyURI")]
        public string URL;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string Type;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string Instance;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string ID;

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string ID_Context;
    }

    // <remarks/>
    [System.Xml.Serialization.XmlRoot(IsNullable = false)]
    public class UniqueID
    {

        // <remarks/>
        [System.Xml.Serialization.XmlAttribute()]
        public string ID;
    }

    // <remarks/>
    [System.Xml.Serialization.XmlRoot(IsNullable = false)]
    public class Provider
    {

        // <remarks/>
        public string Name;

        // <remarks/>
        public string System;

        // <remarks/>
        public string Userid;

        // <remarks/>
        public string Password;
    }

    // <remarks/>
    [System.Xml.Serialization.XmlRoot(IsNullable = false)]
    public class Success
    {
    }
}